lib/inets/src/http_client/httpc_handler.erl

```diff
@@ -622,13 +622,11 @@ do_handle_info({ssl_closed, _}, State = #state{request = undefined}) ->
     {stop, normal, State};
 
 %%% Error cases
-do_handle_info({tcp_closed, _}, #state{session = Session0} = State) ->
-    Socket  = Session0#session.socket,
+do_handle_info({tcp_closed, _}, #state{session = #session{socket = Socket}=Session0} = State) ->
     Session = Session0#session{socket = {remote_close, Socket}},
     %% {stop, session_remotly_closed, State};
     {stop, normal, State#state{session = Session}};
-do_handle_info({ssl_closed, _}, #state{session = Session0} = State) ->
-    Socket  = Session0#session.socket,
+do_handle_info({ssl_closed, _}, #state{session = #session{socket = Socket}=Session0} = State) ->
```

Comment:
Would be nice if we could handle this with the same code in only one function body like this:
```erlang
do_handle_info({Closed, _}, #state{session = #session{socket = Socket}=Session0} = State) when Closed == tcp_closed orelse Closed == ssl_closed ->
      Session = Session0#session{socket = {remote_close, Socket}},
￼    %% {stop, session_remotly_closed, State};
￼    {stop, normal, State#state{session = Session}};
```

Reply:
I merged the PR before I saw these comments.
Please review #6904 which contains the requested changes :)